### DISCUSSION TITLE
Using $refs

### DISCUSSION THREAD

**Discussion by JokerMartini** [quasarframework/quasar#10139] [General - Components / Directives / etc] [answered]
How can I properly use the vue $refs api to display the current time and duration of a video? I feel like i'm doing this properly but it appears to be erroring out. 

I simply just want to display teh current time and duration of the video as it's playing these values should update...

Below is my sample code...
```

<template>
  <q-card class="overflow-hidden">
      {{ videoDuration }}
      {{ videoCurrentTime }}
      <div
        class="video-container"
        style="
          width: 260px;
          height: 140px;
          overflow: hidden;
          background: red;
          position: relative;
        "
      >
        <video
          autoplay
          muted
          loop
          preload
          poster="https://picsum.photos/200/300"
          style="width: 100%; height: 100%"
          ref="previewVideo"
        >
          <source
            src="https://upload.wikimedia.org/wikipedia/commons/transcoded/c/c0/Big_Buck_Bunny_4K.webm/Big_Buck_Bunny_4K.webm.1080p.webm"
            type="video/mp4"
          />
        </video>
      </div>
  </q-card>
</template>

<script>
import { ref, computed, onMounted } from "vue";

export default {
  setup() {

    const previewVideo = ref(null)
    const videoDuration = computed(() => {
      return previewVideo.duration;
    });
    const videoCurrentTime = computed(() => {
      return previewVideo.currentTime;
    });

    return {
      videoDuration,
      videoCurrentTime,
      // var vid = document.getElementById("myVideo");
      // vid.duration
      // vid.currentTime
    };
  },

};
</script>
```

**Comment by hawkeye64** [comment, marked as answer]
It should be `previewVideo.value.duration` and `previewVideo.value.currentTime`, since you need to access refs by value.

**Reply by hawkeye64** [reply]
You should also check your `previewVideo` for null as well, as the computed will run before mounting.

```js
    const videoCurrentTime = computed(() => {
      if (previewVideo.currentTime === null) return ''
      return previewVideo.currentTime;
    });
```

**Reply by JokerMartini** [reply]
Hey @hawkeye64  thank you very much for the info. It appears this didn't resolve the issue, i did as you instructed and no matter what it would return null as a result. The values still appear to return an empty string no matter what, even as the video is playing. I tried to make a fiddle if that helps at all.  The code below should be a plug-in-play setup if you wanted to test.
https://jsfiddle.net/JokerMartini/j7s43bzt/
```

<template>
  <q-card class="overflow-hidden">
      time: {{ videoCurrentTime }} duration: {{ videoDuration }}
      <video
        autoplay
        muted
        loop
        preload
        poster="https://picsum.photos/200/300"
        style="width: 320px; height: 240px"
        ref="previewVideo"
      >
        <source
          src="https://upload.wikimedia.org/wikipedia/commons/transcoded/c/c0/Big_Buck_Bunny_4K.webm/Big_Buck_Bunny_4K.webm.1080p.webm"
          type="video/mp4"
        />
      </video>
  </q-card>
</template>

<script>
import { ref, computed} from "vue";

export default {
  setup() {
    const previewVideo = ref(null);
    const videoDuration = computed(() => {
      if (previewVideo.value === null) return "";
      return previewVideo.value.duration;
    });
    const videoCurrentTime = computed(() => {
      console.log(previewVideo.value);
      if (previewVideo.value === null) return "";
      return previewVideo.value.currentTime;
    });

    return {
      videoDuration,
      videoCurrentTime,
    };
  },
};
</script>
```

**Reply by metalsadman** [reply]
@JokerMartini first fix your fiddle, check logs. use events https://developer.mozilla.org/en-US/docs/Web/HTML/Element/video#events

https://codepen.io/metalsadman/pen/yLboRZa?editors=1010

more info why your fiddle is not working apart from not doing `const { ref, computed } = Vue` or returning your `previewVideo` in your code, read here https://v3.vuejs.org/guide/component-template-refs.html.
specifically the warning 
![image](https://user-images.githubusercontent.com/31038838/126435827-bbc29f5b-0787-42ad-b3ac-6cfc6bc67cb6.png)

more info here https://stackoverflow.com/questions/54927707/why-does-the-vue-documentation-say-refs-are-not-reactive-when-indeed-they-are

**Reply by JokerMartini** [reply]
I appreciate it @metalsadman I was not sure how to best approach what i was trying to do. I didn't know if the refs were my only option or not. Thank you for the added info as well.

**Reply by JokerMartini** [reply]
@metalsadman if i wanted to set the time of the video to 50% (middle of the video) when a user clicks on a button. How would one do that if you are suggesting i do not use $refs? This seems like it would be tricky.
Thanks again
